service/history/workflow/mutable_state_impl.go

```diff
@@ -6132,6 +6174,10 @@ func (ms *MutableStateImpl) closeTransactionUpdateTransitionHistory(
 		ms.GetCurrentVersion(),
 	)
 
+	if len(workflowEventsSeq) == 0 {
+		ms.executionInfo.LastUpdateClock = ms.shard.CurrentVectorClock().GetClock()
+	}
```

Comment:
hmm I realized I can't do this. this field can't be updated after workflow is closed. 
otherwise the ndc vector clock for an old workflow can be larger than the current workflow if the old workflow is keep getting updated after it's closed. 

I will need to figure out something else...

Reply:
ok I changed the logic to only update the field when workflow is running, and renamed this field to LastRunningClock